src/main/java/com/google/cloud/tools/gradle/appengine/appyaml/StageAppYamlExtension.java

```diff
@@ -81,22 +82,37 @@ public void setStagingDirectory(Object stagingDirectory) {
     this.stagingDirectory = project.file(stagingDirectory);
   }
 
+  /** This method is purely for incremental build calculations. */
   @Optional
-  @InputDirectory
-  public List<File> getExtraFilesDirectory() {
-    return extraFilesDirectory;
+  @InputFiles
+  public FileCollection calculateExtraFiles() {
```

Comment:
Ah, so when you annotate a method with `@Input`,`@InputFiles`,`@InputFile`,`@InputDirectory` gradle decides when running a task that it will use those inputs to determine if any input has changed on the task (and decide if it wants to rerun a task or not). It however does not have a method for `@InputDirectories` (plural). So you must manually convert a list of directories to a collection of files for gradle then to take that into account when doing up-to-date checks. `@InputDirectory` already does traversal, but that only works for a single directory. `@InputFiles` doesn't work directory since it wont traverse the directory.

So naming is still that we're pre-calculating something for gradle: `calculateExtraFilesDirectoryInput`?

Reply:
Ah, I see. I think the current naming is OK then.

Reply:
renamed to something more obvious in its intent.